libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -274,7 +288,23 @@ void lua_scripts::reschedule_script(script_info *script) {
 }
 
 void lua_scripts::run(void) {
-    lua_State *L = luaL_newstate();
+    lua_State *L = nullptr;
+
+    // panic should be hooked first
+    if (setjmp(panic_jmp)) {
+        gcs().send_text(MAV_SEVERITY_INFO, "Lua: Caught a panic");
+        if (L != nullptr) {
+            lua_close(L); // shutdown the old state
+        }
+        // remove all the old scheduled scripts
+        for (script_info *script = scripts; script != nullptr; script = script->next) {
+            remove_script(nullptr, script);
```

Comment:
ahh, I see the null check happening here.

Reply:
Yes, but it's probably worth ensuring that it's safe in `remove_script` as well.